service/history/workflow/workflow_task_state_machine.go

```diff
@@ -534,22 +534,17 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskCompletedEvent(
 	}
 
 	// Now write the completed event
-	versionStamp := request.WorkerVersionStamp
-	if !request.UseVersioning {
-		versionStamp = nil
-	}
-	// TODO: omit BinaryChecksum if version stamp is present and matches BinaryChecksum
```

Comment:
can we/should we do anything about this? are we going to have these repeated twice in the event for now?

Reply:
Hmm... I think the worker will send both, let me track this and figure out what we want to do.